Samples.Objc/Samples.ObjC/API/Requests/CreateOrderRequest.m

```diff
@@ -10,35 +10,99 @@
 
 @implementation CreateOrderRequest
 
-- (id)initWith:(OrderIntent)intent purchaseUnits:(NSArray *)purchaseUnits {
+- (id)initWithOrder:(PPCOrderRequest*)order accessToken:(NSString*)token {
   self = [super init];
   if (self) {
-    self.intent = intent;
-    self.purchaseUnits = purchaseUnits;
+    self.order = order;
+    self.accessToken = token;
   }
   return self;
 }
 
 - (NSDictionary *)properties {
+  NSString *intentString = @"";
+  if (self.order.intent == PPCOrderIntentCapture) {
+    intentString = @"CAPTURE";
+  } else {
+    intentString = @"AUTHORIZE";
+  }
 
-  // Convert purchase units to json
   NSMutableArray *purchaseUnitDictionaries = [NSMutableArray new];
-  for (PurchaseUnit *purchaseUnit in [self purchaseUnits]) {
-    NSDictionary *purchaseUnitDictionary = [purchaseUnit dictionaryRepresentation];
+  for (PPCPurchaseUnit *purchaseUnit in self.order.purchaseUnits) {
+    
+    NSMutableArray *itemsDictionary = [NSMutableArray new];
+    
+    for (PPCPurchaseUnitItem *item in purchaseUnit.items) {
+      
+      NSDictionary *itemUnitAmountDictionary = @{
+        @"currency_code": @"USD",
+        @"value": item.unitAmount.value
+      };
+      
+      NSString *itemCategoryString = nil;
+      if (item.category == PPCPurchaseUnitCategoryDigitalGoods) {
+        itemCategoryString = @"DIGITAL_GOODS";
+      } else {
+        itemCategoryString = @"PHYSICAL_GOODS";
+      }
+      
+      NSDictionary *itemDictionary = @{
+        @"name": item.name,
+        @"unit_amount": itemUnitAmountDictionary,
+        @"quantity": item.quantity,
+        @"category": itemCategoryString
+      };
+      
+      [itemsDictionary addObject:itemDictionary];
+    }
+    
+    NSDictionary *breakdownItemTotalDictionary = @{
+      @"currency_code": @"USD",
+      @"value": purchaseUnit.amount.breakdown.itemTotal.value
+    };
+    
+    NSDictionary *breakdownTaxTotalDictionary = @{
+      @"currency_code": @"USD",
+      @"value": purchaseUnit.amount.breakdown.taxTotal.value
+    };
+    
+    NSDictionary *breakdownDictionary = @{
+      @"item_total": breakdownItemTotalDictionary,
+      @"tax_total": breakdownTaxTotalDictionary
+    };
+    
+    NSDictionary *purchaseUnitAmountDictionary = @{
+      @"currency_code": @"USD",
+      @"value": purchaseUnit.amount.value,
+      @"breakdown": breakdownDictionary
+    };
+    
+    NSDictionary *purchaseUnitDictionary = @{
+      @"amount": purchaseUnitAmountDictionary,
+      @"items": itemsDictionary
+    };
```

Comment:
Some of these will result in `nil` exceptions because some of the properties are `@Nullable`:

```
Thread 1: "*** -[__NSPlaceholderDictionary initWithObjects:forKeys:count:]: attempt to insert nil object from objects[1]"
```

```objectivec
if (purchaseUnit.amount.breakdown) {
  // Then add breakdown
}
```

Thoughts?

Reply:
ah this is nice!!!!! I'll add this :smile: